sys/kern/file_syscalls.c

```diff
@@ -33,6 +33,13 @@ int do_write(proc_t *p, int fd, uio_t *uio) {
 
   uio->uio_ioflags |= f->f_flags & IO_MASK;
   error = f->f_ops->fo_write(f, uio);
+  if (error) {
+    if (error == EPIPE) {
+      mtx_lock(&p->p_lock);
```

Comment:
I prefer `proc_lock` or `WITH_PROC_LOCK`.

Reply:
thx 4 suggestion, It was really hard for me to find correct way of using mutexes in kernel